src/promptflow-tools/promptflow/tools/common.py

```diff
@@ -10,15 +10,15 @@
 from promptflow.tools.exception import ChatAPIInvalidRole, WrappedOpenAIError, LLMError, JinjaTemplateError, \
     ExceedMaxRetryTimes, ChatAPIInvalidFunctions, FunctionCallNotSupportedInStreamMode, \
     ChatAPIFunctionRoleInvalidFormat
-from typing import Set
+from typing import List
 
 
-def validate_role(role: str, valid_roles: Set[str] = None):
+def validate_role(role: str, valid_roles: List[str] = None):
```

Comment:
Why are we using a list instead of a set? Wouldn't a set be more efficient?

Reply:
It's 4 items, and list maintains order.